File: Exercícios/entre12e20.c
#include <stdio.h>
#include <locale.h>
#include <stdbool.h>

int main()
{
    setlocale(LC_ALL, ("pt-br"));

    int numero_digitado;

    while (true)
    {
        printf("\nInsira um número: ");
        scanf("%d", &numero_digitado);

        if (numero_digitado >= 12 && numero_digitado <= 20)
        {
            printf("\n%d\n", numero_digitado);
        }
        else
        {
            printf("\nentrada inválida\n");
        }
    }

    return 0;
}

File: Exercícios/entre12e20.java
import java.util.Scanner;

public class Main {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        while (true) {
            System.out.print("\nInsira um número: ");
            int numeroDigitado = scanner.nextInt();

            if (numeroDigitado >= 12 && numeroDigitado <= 20) {
                System.out.println("\n" + numeroDigitado);
            } else {
                System.out.println("\nEntrada inválida");
            }
        }
    }
}